public/app/plugins/datasource/graphite/configuration/ConfigEditor.tsx

```diff
@@ -1,20 +1,99 @@
-import { GraphiteDetails } from './GraphiteDetails';
-import { GraphiteOptions } from '../types';
-
-export const ConfigEditor = (props: DataSourcePluginOptionsEditorProps<GraphiteOptions>) => {
-  const { options, onOptionsChange } = props;
-
-  return (
-    <>
-      <DataSourceHttpSettings
-        defaultUrl="http://localhost:8080"
-        dataSourceConfig={options}
-        onChange={onOptionsChange}
-      />
-      <GraphiteDetails value={options} onChange={onOptionsChange} />
-    </>
-  );
-};
+import React, { PureComponent } from 'react';
+import { DataSourceHttpSettings, FormLabel, Button } from '@grafana/ui';
+import { DataSourcePluginOptionsEditorProps, onUpdateDatasourceJsonDataOptionSelect } from '@grafana/data';
+import { GraphiteOptions, GraphiteType } from '../types';
+import { Select } from '@grafana/ui/src/components/Forms/Select/Select';
+
+const graphiteVersions = [
+  { label: '0.9.x', value: '0.9' },
+  { label: '1.0.x', value: '1.0' },
+  { label: '1.1.x', value: '1.1' },
+];
+
+const graphiteTypes = Object.keys(GraphiteType).map((key: string) => ({
+  label: key,
+  value: (GraphiteType as any)[key],
+}));
+
+export type Props = DataSourcePluginOptionsEditorProps<GraphiteOptions>;
+
+interface State {
+  showMetricTankHelp: boolean;
+}
+
+export class ConfigEditor extends PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      showMetricTankHelp: false,
+    };
+  }
+
+  render() {
+    const { options, onOptionsChange } = this.props;
+    const { showMetricTankHelp } = this.state;
+
+    const currentVersion =
+      graphiteVersions.find(item => item.value === options.jsonData.graphiteVersion) ?? graphiteVersions[2];
+
+    return (
+      <>
+        <DataSourceHttpSettings
+          defaultUrl="http://localhost:8080"
+          dataSourceConfig={options}
+          onChange={onOptionsChange}
+        />
+        <h3 className="page-heading">Graphite details</h3>
+        <div className="gf-form-group">
+          <div className="gf-form">
+            <FormLabel tooltip="This option controls what functions are available in the Graphite query editor.">
+              Version
+            </FormLabel>
+            <Select
+              value={currentVersion}
+              options={graphiteVersions}
+              width={8}
+              onChange={onUpdateDatasourceJsonDataOptionSelect(this.props, 'graphiteVersion')}
+            />
+          </div>
+          <div className="gf-form-inline">
+            <FormLabel>Type</FormLabel>
+            <Select
+              options={graphiteTypes}
+              value={graphiteTypes.find(type => type.value === options.jsonData.graphiteType)}
+              width={8}
+              onChange={onUpdateDatasourceJsonDataOptionSelect(this.props, 'graphiteType')}
+            />
+
+            <Button
+              style={{ marginLeft: '8px', marginTop: '5px' }}
```

Comment:
I would avoid inline styling.  I believe the standard tool here for custom styling/css is emotion.

Reply:
This was already done in master. I am not 100% on why. Can investigate though. @scottlepp